internal/controller/operatorconfigmap_controller.go

```diff
@@ -950,3 +959,69 @@ func (c *OperatorConfigMapReconciler) deleteDelegatedCSI() error {
 	}
 	return nil
 }
+
+func (c *OperatorConfigMapReconciler) removeNoobaaOperator() error {
+
+	nbList := &nbv1.NooBaaList{}
+	err := c.list(nbList)
+	if err != nil && !kerrors.IsNotFound(err) {
+		return fmt.Errorf("failed to list noobaa CRs: %v", err)
+	}
+	if len(nbList.Items) != 0 {
+		for i := range nbList.Items {
+			for _, ref := range nbList.Items[i].GetOwnerReferences() {
+				if ref.Kind == "StorageClient" {
+					if err := c.delete(&nbList.Items[i]); err != nil {
+						return err
+					}
+				}
+			}
+		}
+	}
+
+	csvList := &opv1a1.ClusterServiceVersionList{}
+	if err := c.list(csvList); err != nil {
+		return fmt.Errorf("failed to list csvs: %v", err)
+	}
+	csv := utils.Find(
+		csvList.Items,
+		func(csv *opv1a1.ClusterServiceVersion) bool {
+			return strings.HasPrefix(csv.Name, "mcg-operator") || strings.HasPrefix(csv.Name, "noobaa-operator")
+		},
+	)
+	if csv != nil {
+		if !isOwnedByOdfSubscription(csv) {
+			if err := c.delete(csv); err != nil && !kerrors.IsNotFound(err) {
+				c.log.Error(err, "failed to delete noobaa operator csv")
+				return err
+			}
+		}
+	}
+	noobaaSubscription, err := c.getSubscriptionByPackageName("mcg-operator")
+	if kerrors.IsNotFound(err) {
+		noobaaSubscription, err = c.getSubscriptionByPackageName("noobaa-operator")
+		if kerrors.IsNotFound(err) {
+			return nil
+		} else if err != nil {
+			return err
+		}
+	} else if err != nil {
+		return err
+	}
```

Comment:
pls fix this error handling, if sub is found and odf-op is the owner delete it, it's not an error if sub isn't found.